x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeService.java

```diff
@@ -311,7 +311,7 @@ static final class TransportRemoteSink implements RemoteSink {
 
         @Override
         public void fetchPageAsync(boolean allSourcesFinished, ActionListener<ExchangeResponse> listener) {
-            final long reservedBytes = estimatedPageSizeInBytes.get();
+            final long reservedBytes = allSourcesFinished ? 0 : estimatedPageSizeInBytes.get();
```

Comment:
Unrelated to this PR, but this allows for faster cleaning of exchange sinks in case of circuit-breaking exceptions.